contracts/VestingSwap.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity >=0.6.6;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../interfaces/ISaleHybridToken.sol";
+import "../interfaces/IPresale.sol";
+import "../libraries/SafeTransfer.sol";
+import "../libraries/PresaleConstants.sol";
+
+
+contract VestingSwap is Ownable {
+    using SafeMath for uint256;
+
+    bytes32 public alphaPresale = keccak256("alphaPresale");
+    bytes32 public betaPresale = keccak256("betaPresale");
+    bytes32 public gammaPresale = keccak256("gammaPresale");
+
+    uint32 private constant SECONDS_PER_DAY = 24 * 60 * 60;
+    uint32 private constant SECONDS_PER_MONTH = SECONDS_PER_DAY * 30;
+
+    address private HBT;
+
+    struct swapInfo {
+        address addr;
+        uint limit;
+        uint[] vesting;
+        uint start;
+        uint swaped;
+    }
+
+    mapping(bytes32 => swapInfo) public swap;
+    mapping(address => uint) public swapedAmountOf;
+
+    event Swap(address account, uint amount);
+
+    modifier isStarted(bytes32 _presale) {
+        require(swap[_presale].start > 0);
+        _;
+    }
+
+    constructor(address _alphaPresale, address _betaPresale, address _gammaPresale, address _HBT) public {
+        HBT = _HBT;
+
+        swap[alphaPresale].addr = _alphaPresale;
```

Comment:
Please, explain, why you need `bytes32 public betaPresale = keccak256("betaPresale");` here.
Wouldn't making `mapping(bytes32 => swapInfo) public swap;` into `mapping(address => swapInfo) public swap;`   would simplify the call structure and general architecture?